torch/csrc/autograd/engine.h

```diff
@@ -266,6 +269,12 @@ struct TORCH_API Engine {
  std::shared_ptr<ThreadPoolShared> thread_pool_shared_;
 
 private:
+  // Number of non-reentrant threads
+  std::atomic<uint32_t> non_reentrant_thread_count_;
```

Comment:
nit: this does not need to be atomatic as all the changes are either in std::cal_once or protected by the lock below.